examples/06_vol_geom_single_file/main.c

```diff
@@ -0,0 +1,183 @@
+  } else {
+    gfx_update_mesh_from_mem( mesh_ptr, points_ptr, 3, uvs_ptr, 2, NULL, 3, NULL, 0, indices_type, n_vertices, true );
+  }
+
+  if ( texture_ptr && vols_info_ptr->hdr.textured ) {
+    uint8_t* vols_texture_ptr = (uint8_t*)&vols_frame_data.block_data_ptr[vols_frame_data.texture_offset];
+    size_t vols_texture_sz    = vols_frame_data.texture_sz;
+    if ( !vol_basis_transcode( 3, vols_texture_ptr, vols_texture_sz, output_blocks_ptr, OUTPUT_DIMS * OUTPUT_DIMS, &texture_ptr->w, &texture_ptr->h ) ) {
+      fprintf( stderr, "ERROR transcoding image %i failed\n", frame_number );
+      return 1;
+    }
+    _update_compressed_texture( 0, output_blocks_ptr, texture_ptr );
+  }
+  return true;
+}
+
+int main( int argc, char** argv ) {
+  const char* filename_vols = argv[1];
+  if ( argc < 2 ) {
+    filename_vols = "../samples/combined.vols";
+    printf( "Usage: %s MYFILE.vols\n. Using default: %s", argv[0], filename_vols );
+  }
+
+  output_blocks_ptr = (uint8_t*)malloc( OUTPUT_DIMS * OUTPUT_DIMS );
+  if ( !vol_basis_init() ) {
+    fprintf( stderr, "ERROR vol_basis_init() failed\n" );
+    return 1;
+  }
+  printf( "Loading combined header&sequence file `%s`\n", filename_vols );
+
+  // Populate the meta-data struct from a single .vols file.
+  vol_geom_info_t vols_info = ( vol_geom_info_t ){ .biggest_frame_blob_sz = 0 };
+  if ( !vol_geom_create_file_info_from_file( filename_vols, &vols_info ) ) {
+    fprintf( stderr, "ERROR vol_geom_create_file_info_from_file.\n" );
+    return 1;
+  }
+
+  gfx_start( "vol_geom Single Vols File OpenGL Example\n", 512, 512, false );
+  gfx_texture_t compressed_texture;
+  {
+    GLuint tex_cmpr1 = 0;
+    glGenTextures( 1, &tex_cmpr1 );
+    GLenum internal_format = GL_COMPRESSED_RGBA_S3TC_DXT5_EXT;
+    glBindTexture( GL_TEXTURE_2D, tex_cmpr1 );
+    glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE );
+    glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE );
+    glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST );
+    glTexParameteri( GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST );
+    glCompressedTexImage2D( GL_TEXTURE_2D, 0, internal_format, OUTPUT_DIMS, OUTPUT_DIMS, 0, OUTPUT_DIMS * OUTPUT_DIMS, NULL );
+    glBindTexture( GL_TEXTURE_2D, 0 );
+    compressed_texture.w = compressed_texture.h = OUTPUT_DIMS;
+    compressed_texture.handle_gl                = tex_cmpr1;
+    compressed_texture.n_channels               = 4;
+    memset( &compressed_texture.properties, 0, sizeof( gfx_texture_properties_t ) );
+  }
+  gfx_mesh_t mesh = gfx_create_mesh_from_mem( NULL, 3, NULL, 2, NULL, 3, NULL, 0, 1, 0, true );
+  if ( !_update_mesh_with_frame( &mesh, 0, filename_vols, &vols_info, &compressed_texture ) ) {
+    fprintf( stderr, "update mesh with frame failed\n" );
+    return 1;
+  }
+
+  if ( vols_info.hdr.audio == 1 ) { // 0=no audio, 1=ogg vorbis/opus
```

Comment:
Should we change this to mp3 as this is what we are going to use?

Reply:
thanks. just changed to a generic statement for now. i would like to make the demo play audio but can move that into a subsequent PR to break it up into smaller pieces.